### PR TITLE
Clean/reactive

### PR DESCRIPTION
**Pull request by jypuigbo**
Cleaned reactiveLayer branch.

### CODE REVIEW

File: app/reactiveLayer/allostaticController/conf/default.ini [hunk 1]
```diff
@@ -7,31 +7,19 @@ exploration-decay                   0.01
 exploration-priority                1
 exploration-under-behavior          tagging
 exploration-under-behavior-port     /BehaviorManager/trigger:i
-exploration-sensation-port          /opcSensation/opc_has_unknown:o
+exploration-key          			unknown
```
Comment (Tobias-Fischer) [inline app/reactiveLayer/allostaticController/conf/default.ini]: Tabs -> spaces

File: app/reactiveLayer/allostaticController/conf/default.ini [hunk 2]
```diff
@@ -7,31 +7,19 @@ exploration-decay                   0.01
...
-touch-after-trigger           ((unfreeze all) (par touch val 0.5))
-
 demonstration-homeostasisMin          0.25
 demonstration-homeostasisMax          0.75
 demonstration-decay                   0.004
 demonstration-priority                1
 demonstration-under-behavior          pointing
 demonstration-under-behavior-port     /BehaviorManager/trigger:i
-demonstration-sensation-port          /opcSensation/opc_has_known:o
+demonstration-key			          known
```
Comment (Tobias-Fischer) [inline app/reactiveLayer/allostaticController/conf/default.ini]: Tabs -> spaces

File: src/modules/ears/src/ears.cpp [hunk 3]
```diff
@@ -194,26 +194,37 @@ bool ears::updateModule() {
...
             sObject = bSemantic.check("object", Value("none")).asString();
-            sCommand = "followingOrder";
+            //sCommand = "followingOrder";
         } else if(sQuestionKind == "SENTENCEBODYPART") {
             sAction = bSemantic.check("predicateBodypart", Value("none")).asString();
             sObject = bSemantic.check("bodypart", Value("none")).asString();
             sObjectType = "bodypart";
             sCommand = "followingOrder";
-        } else if(sQuestionKind == "SENTENCENARRATIVE") {
+        }/* else if(sQuestionKind == "SENTENCENARRATIVE") {
```
Comment (Tobias-Fischer) [inline src/modules/ears/src/ears.cpp]: Delete commented out parts.

File: src/modules/ears/src/ears.cpp [hunk 4]
```diff
@@ -248,12 +259,12 @@ bool ears::updateModule() {
...
             sObjectType = "end";
             sObject = "";
-        } else {
+        }*/ else {
             yError() << "[ears] Unknown predicate: " << sQuestionKind;
             return true;
         }
         //send rpc data to planner
-        if (onPlannerMode) {
+        /*if (onPlannerMode) {
```
Comment (Tobias-Fischer) [inline src/modules/ears/src/ears.cpp]: Delete commented out parts.

File: src/modules/reactiveLayer/allostaticController/src/allostaticController.cpp [hunk 5]
```diff
@@ -394,27 +375,27 @@ bool AllostaticController::updateAllostatic()
 
     // Create relation for the drive
 
-    iCub->opc->addOrRetrieveEntity<Agent>("icub");
+    /*iCub->opc->addOrRetrieveEntity<Agent>("icub");
```
Comment (Tobias-Fischer) [inline src/modules/reactiveLayer/allostaticController/src/allostaticController.cpp]: Delete commented out parts.

**Comment by jypuigbo** [comment]
Indeed, I removed `recognitionOrder` without noticing. It's put back again. I'll check what you say about the documentation. 

All the comments have been addressed. 
The old file `internalVariablesDecay.h` which was not used has been removed as well. 

If you want, accept this when ready and I'll do the changes in the documentation directly in master.

File: app/reactiveLayer/behaviorManager/conf/default.ini [hunk 6]
```diff
@@ -2,7 +2,7 @@ use_ears        false
 
 [BEHAVIORS]
 
-behaviors	(tagging, pointing, moveObject, ask, followingOrder, recognitionOrder)
+behaviors	(tagging, pointing, moveObject)
 ttsOptions      "iCub"
 
 [followingOrder]
```
Comment (Tobias-Fischer) [inline app/reactiveLayer/behaviorManager/conf/default.ini]: Hey @jypuigbo,
One last question before merging. Should `followingOrder` really be removed? Will we still be able to say "Point to the octopus", where octopus is an unknown object? Is this handled by `tagging`? If not, we should keep `followingOrder`
If `followingOrder` is to be removed, we should remove all references to it, i.e. the `[followingOrder]` group in `app/reactiveLayer/behaviorManager/conf/default.ini` and the reference to "followingOrder" in `ears.cpp` Line 221/226.

Best, Tobi

Comment (jypuigbo) [inline app/reactiveLayer/behaviorManager/conf/default.ini]: I added `recognitionOrder` to the config file. 
`FollowingOrder` is not needed anymore. It was not used during the WYSIWYD review, already. 
I'm commiting the related changes.

Comment (jypuigbo) [inline app/reactiveLayer/behaviorManager/conf/default.ini]: Sorry, do we keep the option to ask "what did you learn from the arm babbling?" ? In that case I'll have to add this as a behavior (should be done quickly).

**Comment by jypuigbo** [comment]
Your changes look good to me. Only check my last reply to your inline comment and I react to your answer.

File: app/reactiveLayer/behaviorManager/conf/default.ini (hunk 6 above)
Comment (Tobias-Fischer) [inline app/reactiveLayer/behaviorManager/conf/default.ini]: Nah, we can remove. All good.

**Comment by Tobias-Fischer** [comment]
Hi @jypuigbo,
all looks good to me know. Could you please check https://github.com/robotology/icub-client/pull/36/commits/1f1b15a3ae04d9cd76e525dca52c32032492011c and tell me whether this is correct? If so, we can merge :)

**Comment by jypuigbo** [comment]
Yes, they are correct.

**Comment by Tobias-Fischer** [comment]
Great, thanks, merged :)

**Pull request merged**